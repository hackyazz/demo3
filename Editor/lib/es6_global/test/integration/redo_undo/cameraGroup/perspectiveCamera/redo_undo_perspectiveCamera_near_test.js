

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as RedoUndoTool$WonderEditor from "../../tool/RedoUndoTool.js";
import * as GameObjectTool$WonderEditor from "../../../../tool/GameObjectTool.js";
import * as EventListenerTool$WonderEditor from "../../../../unit/tool/EventListenerTool.js";
import * as DirectorToolEngine$WonderEditor from "../../../../tool/engine/DirectorToolEngine.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as BuildComponentForCurryTool$WonderEditor from "../../../../tool/BuildComponentForCurryTool.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../../../tool/engine/NoWorkerJobConfigToolEngine.js";
import * as MainEditorCameraProjectionTool$WonderEditor from "../../../inspector/composable_component/sceneTree_inspector/cameraGroup/cameraProjection/tool/MainEditorCameraProjectionTool.js";

describe("redo_undo: perspectiveCamera near", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        var _changeNear = function (value) {
          return MainEditorCameraProjectionTool$WonderEditor.changeNearAndBlur(GameObjectTool$WonderEditor.getCurrentGameObjectPerspectiveCamera(/* () */0), value, undefined, undefined, /* () */0);
        };
        var _simulateTwiceChangeNear = function () {
          _changeNear(10.112);
          return _changeNear(123.12);
        };
        var _beforeEach = function () {
          MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, undefined, undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
          Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
          MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setSceneFirstCameraToBeCurrentSceneTreeNode);
          return DirectorToolEngine$WonderEditor.prepareAndInitAllEnginState(/* () */0);
        };
        return RedoUndoTool$WonderEditor.testRedoUndoTwoStep(sandbox, "prepare first step: set currentSceneTreeNode", /* tuple */[
                    _simulateTwiceChangeNear,
                    _beforeEach,
                    (function () {
                        return /* () */0;
                      })
                  ], BuildComponentForCurryTool$WonderEditor.buildCameraProjection);
      }));

export {
  
}
/*  Not a pure module */
