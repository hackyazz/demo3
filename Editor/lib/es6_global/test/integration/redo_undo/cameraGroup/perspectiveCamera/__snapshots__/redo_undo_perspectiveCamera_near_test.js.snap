// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-camera-projection"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          perspective
        </option>
      </select>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Near
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Far
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="50000"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Fovy
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-camera-projection"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          perspective
        </option>
      </select>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Near
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Far
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="50000"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Fovy
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-camera-projection"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          perspective
        </option>
      </select>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Near
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Far
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="50000"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Fovy
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-camera-projection"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          perspective
        </option>
      </select>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Near
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Far
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="50000"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Fovy
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-camera-projection"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          perspective
        </option>
      </select>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Near
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Far
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="50000"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Fovy
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </div>
  </article>
</article>
`;
